core/src/main/scala/com/salesforce/op/stages/impl/feature/DateToUnitCircleTransformer.scala

```diff
@@ -60,31 +77,122 @@ import org.joda.time.{DateTime => JDateTime}
+ *
+ * parameter timePeriod The time period to extract from the timestamp
+ * enum from: DayOfMonth, DayOfWeek, DayOfYear, HourOfDay, MonthOfYear, WeekOfMonth, WeekOfYear
+ *
+ * We extract the timePeriod from the timestamp and
+ * map this onto the unit circle containing the number of time periods equally spaced.
+ * For example, when timePeriod = HourOfDay, the timestamp 01/01/2018 6:37 maps to the point on the circle with
+ * angle radians = 2*math.Pi*6/24
+ * We return the cartesian coordinates of this point: (math.cos(radians), math.sin(radians))
+ *
+ * The first time period always has angle 0.
+ *
+ * Note: We use the ISO week date format https://en.wikipedia.org/wiki/ISO_week_date#First_week
+ * Monday is the first day of the week
+ * & the first week of the year is the week wit the first Monday after Jan 1.
+ */
+class DateMapToUnitCircleVectorizer[T <: DateMap]
+(
+  uid: String = UID[DateMapToUnitCircleVectorizer[_]]
+)(implicit tti: TypeTag[T], override val ttiv: TypeTag[T#Value]) extends SequenceEstimator[T, OPVector](
+  operationName = "dateMapToUnitCircle",
+  uid = uid
+) with DateToUnitCircleParams with MapVectorizerFuns[Long, T]  {
 
-  final def setTimePeriod(value: TimePeriod): this.type = set(timePeriod, value.entryName)
+  override def makeVectorMetadata(allKeys: Seq[Seq[String]]): OpVectorMetadata = {
+    val meta = vectorMetadataFromInputFeatures
+    val timePeriod = getTimePeriod
+
+    val cols = for {
+      (keys, col) <- allKeys.zip(meta.columns)
+      key <- keys
+      dec <- DateToUnitCircle.metadataValues(timePeriod)
+    } yield new OpVectorColumnMetadata(
+      parentFeatureName = col.parentFeatureName,
+      parentFeatureType = col.parentFeatureType,
+      grouping = Option(key),
+      descriptorValue = Option(dec)
+    )
 
-  override def transformFn: T => OPVector = timestamp => {
-    val datetime: Option[JDateTime] = timestamp.value.map(new JDateTime(_))
-    val (timePeriod, periodSize) = getTimePeriod match {
+    meta.withColumns(cols.toArray)
+  }
+
+  override def fitFn(dataset: Dataset[Seq[Map[String, Long]]]): SequenceModel[T, OPVector] = {
+    val shouldClean = $(cleanKeys)
+    val allKeys = getKeyValues(dataset, shouldClean, shouldCleanValues = false)
+
+    val meta = makeVectorMetadata(allKeys)
+    setMetadata(meta.toMetadata)
+    new DateMapToUnitCircleVectorizerModel[T](allKeys = allKeys, shouldClean = shouldClean,
+      timePeriodString = getTimePeriod.entryName, operationName = operationName, uid = uid
+    )
+  }
+
+}
+
+final class DateMapToUnitCircleVectorizerModel[T <: DateMap] private[op]
+(
+  val allKeys: Seq[Seq[String]],
+  val shouldClean: Boolean,
+  val timePeriodString: String,
+  operationName: String,
+  uid: String
+)(implicit tti: TypeTag[T]) extends SequenceModel[T, OPVector](operationName = operationName, uid = uid)
+  with CleanTextMapFun {
+
+  private val timePeriod: TimePeriod = TimePeriod.withNameInsensitive(timePeriodString)
```

Comment:
please move `TimePeriod` enum to `features` subproject and register to `OpPipelineStageReadWriteShared.formats`, then you will be able to pass `val timePeriod: TimePeriod` as ctor argument for `DateMapToUnitCircleVectorizerModel`. (see how it's done with `HashSpaceStrategy`)